### PR TITLE
First step towards fixing 688

### PR DESCRIPTION
**Pull request by hkaiser**
- renamed all functions `id_type get_gid()` to `get_id`
- applied backwards compatibility flag `HPX_HAVE_COMPONENT_GET_GID_COMPATIBILITY` to temporarily still support `get_gid`

### CODE REVIEW

File: hpx/runtime/actions/continuation.hpp [hunk 1]
```diff
@@ -172,10 +172,17 @@ namespace hpx { namespace actions
...
         naming::id_type const& get_gid() const
         {
             return gid_;
         }
+#endif
+
+        naming::id_type const& get_id() const
+        {
+            return gid_;
+        }
```
Comment (sithhell) [inline hpx/runtime/actions/continuation.hpp]: Does a continuation also need a `get_unmanaged_id` function?

File: hpx/runtime/components/client_base.hpp [hunk 2]
```diff
@@ -289,11 +289,18 @@ namespace hpx { namespace components
...
         {
             return gid_.get();
         }
 
+#endif
+        naming::id_type const & get_id() const
+        {
+            return gid_.get();
+        }
+
```
Comment (sithhell) [inline hpx/runtime/components/client_base.hpp]: Do we also want to have a `get_unmanaged_id` function here?

File: hpx/runtime/components/runtime_support.hpp [hunk 3]
```diff
@@ -168,7 +168,7 @@ namespace hpx { namespace components
         }
 
         ///////////////////////////////////////////////////////////////////////
-        naming::id_type const& get_gid() const
+        naming::id_type const& get_id() const
```
Comment (sithhell) [inline hpx/runtime/components/runtime_support.hpp]: Are we sure this is not used externally anywhere?

File: hpx/runtime/actions/continuation.hpp (hunk 1 above)
Comment (hkaiser) [inline hpx/runtime/actions/continuation.hpp]: Why should it? This was renamed for consistency reasons only. The user will never see or call this function directly anyways.

File: hpx/runtime/components/client_base.hpp (hunk 2 above)
Comment (hkaiser) [inline hpx/runtime/components/client_base.hpp]: Hmm, the id stored in a client could be both, managed or unmanaged, we don't know this. Does the user ever want to retrieve an id which is guaranteed to be unmanaged? Do you have an use case?

File: hpx/runtime/components/runtime_support.hpp (hunk 3 above)
Comment (hkaiser) [inline hpx/runtime/components/runtime_support.hpp]: I thought I was sure about that, but I will wrap it into the pp constant.

**Comment by hkaiser** [comment]
All review comments have been addressed now.

File: hpx/runtime/actions/continuation.hpp (hunk 1 above)
Comment (sithhell) [inline hpx/runtime/actions/continuation.hpp]: In that case we can remove the macro compatibility macro already, no?

Comment (hkaiser) [inline hpx/runtime/actions/continuation.hpp]: Uhhh. Again - for compatibility reasons I'd suggest to leave it in.

File: hpx/runtime/components/client_base.hpp (hunk 2 above)
Comment (sithhell) [inline hpx/runtime/components/client_base.hpp]: The usecase is, for example, a client that the user guarantees to keep alive, for that, he can use an unmanaged id for async calls. However, the real problem is if the client gets constructed from an unmanaged id to begin with, then this whole debate is meaningless after all.

**Comment by sithhell** [comment]
Thanks, looking OK now.

File: hpx/runtime/components/client_base.hpp (hunk 2 above)
Comment (hkaiser) [inline hpx/runtime/components/client_base.hpp]: On one hand, you are proposing to get rid of id_types altogether, and now you're suggesting to expose unmanaged id_types to the user (for questionable optimizations). That is a contradiction which does not make any sense to me, sorry.

Comment (sithhell) [inline hpx/runtime/components/client_base.hpp]: This is completely orthogonal to getting rid of id_type. We have them right now, that's a fact, this PR is dealing with an API related to id_type, not getting rid of it. Unfortunately, `get_id` in client_base could return either a managed or unmanaged `id_type`. But that's probably OK for now.
